pf_driver/src/ros/scan_publisher.cpp

```diff
@@ -83,6 +85,8 @@ void ScanPublisher::to_msg_queue(T &packet, uint16_t layer_idx)
         else
             data = packet.distance[i] / 1000.0;
         msg->ranges[idx + i] = std::move(data);
+        if(packet.amplitude[i] >= 32 && !std::is_same<T,PFR2000Packet_A>::value)
```

Comment:
I am not comfortable checking for a specific packet type (`std::is_same<T,PFR2000Packet_A>::value`). Instead we could check for size of intensities 
```
if(!packet.amplitude.empty() && packet.amplitude[i] >= 32)
```

Reply:
Makes sense.  Are you okay with referring to the specific packet type above?
https://github.com/PepperlFuchs/ROS_driver/blob/9a60a7c3a08901b55744813d20b7a9ab86524215/pf_driver/src/ros/scan_publisher.cpp#L67-L68

Reply:
This line can be removed (?) so that `amplitude` is empty:
https://github.com/PepperlFuchs/ROS_driver/blob/64e3620341323e6fac48a3259e5de55feeb60a87/pf_driver/src/pf/pf_packet.cpp#L55
And you could check this:
```
if(!packet.amplitude.empty() && packet.amplitude[i] >= 32)
```
So that we are not referring to any specific packet type within the function. Would that be ok?

Reply:
Ah, yeah, I think the bit in `pf_packet.cpp` should do it.